server/utils/getFilteredEntries.js

```diff
@@ -2,10 +2,14 @@ const { getPluginService } = require("../utils/getPluginService");
 
 const buildDbQuery = (locale, model) => {
   const { contentTypes } = getPluginService(strapi, "settingsService").get();
-  const { whereConstraints } = contentTypes.find(entry => entry.uid === model);
-  if (locale) return { where: { locale: locale,...whereConstraints } };
-
-  return {};
+  const { queryConstraints } = contentTypes.find(entry => entry.uid === model);
+  if (locale) {
+    if (queryConstraints.where) {
+      return {...queryConstraints, where: { ...queryConstraints.where, locale: locale, } };
+    }
+    return {...queryConstraints, where: {  locale: locale, } };
+  }
+  return {...queryConstraints};
```

Comment:
Please squash the nested if returns into a single return and fix the server throwing a 500 if no queryConstraints have been set for a content type.

I think something like this should work:
```
return {
    ...queryConstraints,
    where: {
      ...(queryConstraints?.where && { ...queryConstraints.where }),
      ...(locale && { locale: locale }),
    },
  };
```